game-core/src/test/java/games/strategy/engine/lobby/PlayerEmailValidationTest.java

```diff
@@ -0,0 +1,37 @@
+package games.strategy.engine.lobby;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+import java.util.Arrays;
+import java.util.Collections;
+import org.junit.jupiter.api.Test;
+
+class PlayerEmailValidationTest {
+  @Test
+  void shouldReturnTrueWhenAddressIsValid() {
+    Arrays.asList(
+            "some@some.com",
+            "some.someMore@some.com",
+            "some@some.com some2@some2.com",
+            "some@some.com some2@some2.co.uk",
+            "some@some.com some2@some2.co.br",
+            "",
+            "some@some.some.some.com")
```

Comment:
I know the tests are just copied but JUnit has a maven artifact that allows for parameterized tests.
It works actually really well and makes this probably more readable

Reply:
This is in part habit on my part; I've yet to really get used to parameterized tests and they have their limitations. Some early versions I really did not like very much and thought they made things less readable. AFAIK the latest does need some magic glue to work.

Is it worth fixing this as a one-off, or would we be better to convert many unit tests to be parameterized all in one go? I'm not sure i like the idea of further increasing scope and mixing more changes into this update, particularly when we are no worse off after the update.

Reply:
I'm fine with the latter 🤷‍♂

Reply:
Well using the JUnit 5 variant you basically have a couple of options on how you want to retrieve the parameters for the test.
I believe there's an option for an external csv file, an option for "inline csv" which might be what you want, or an option to specify a source method, unfortunately via reflection because annotations don't allow lambdas or any other potentially mutable object.

Reply:
BUT IntelliJ supports the reflection variant and actually recognizes that this parameterized test is feeded by the other method.

Reply:
CSV file for this one test is not a good option IMO. It's just one test, I don't think it would scale well as we should wind up with thousands more unit tests. Inline as well, why go to the complication and the formatting headache?

We could have a 'feeder' method, it does have limitations, I'm not necessarily convinced it's the right thing to do and would not want to couple it with more changes. The main change here is to extract the validation code so it can be used elsewhere. Really the only update to all this is one new line in ServerQuarantine that does additional validation.

Reply:
For brand new code, I would agree we should look at test parameterization, particularly 'feeder' methods. That does add a layer of indirection and non-locality, which has me a bit pressed to be 100% convinced it's actually better. I do think it's a valid thing to bring up, so I appreciate the consideration/conversation of it.